src/webhooks/registry.ts

```diff
@@ -354,11 +356,71 @@ export interface WebhookProcessParams extends AdapterArgs {
   rawBody: string;
 }
 
+const statusTextLookup: {[key: string]: string} = {
+  [StatusCode.Ok]: 'OK',
+  [StatusCode.BadRequest]: 'Bad Request',
+  [StatusCode.Unauthorized]: 'Unauthorized',
+  [StatusCode.NotFound]: 'Not Found',
+  [StatusCode.InternalServerError]: 'Internal Server Error',
+};
+
+function checkWebhookRequest(
+  rawBody: string,
+  headers: Headers,
+): {
+  webhookOk: boolean;
+  errorMessage: string;
+  hmac: string;
+  topic: string;
+  domain: string;
+} {
+  const retVal = {
+    webhookOk: true,
+    errorMessage: '',
+    hmac: '',
+    topic: '',
+    domain: '',
+  };
+
+  if (rawBody.length) {
+    const hmac = getHeader(headers, ShopifyHeader.Hmac);
+    const topic = getHeader(headers, ShopifyHeader.Topic);
+    const domain = getHeader(headers, ShopifyHeader.Domain);
+
+    const missingHeaders: ShopifyHeader[] = [];
+    if (!hmac) {
+      missingHeaders.push(ShopifyHeader.Hmac);
+    }
+    if (!topic) {
+      missingHeaders.push(ShopifyHeader.Topic);
+    }
+    if (!domain) {
+      missingHeaders.push(ShopifyHeader.Domain);
+    }
+
+    if (missingHeaders.length) {
+      retVal.webhookOk = false;
+      retVal.errorMessage = `Missing one or more of the required HTTP headers to process webhowebhookOks: [${missingHeaders.join(
+        ', ',
+      )}]`;
+    } else {
+      retVal.hmac = hmac as string;
+      retVal.topic = topic as string;
+      retVal.domain = domain as string;
+    }
+  } else {
+    retVal.webhookOk = false;
+    retVal.errorMessage = 'No body was received when processing webhook';
+  }
+
+  return retVal;
+}
+
 export function createProcess(config: ConfigInterface) {
   return async function process({
     rawBody,
     ...adapterArgs
-  }: WebhookProcessParams): Promise<NormalizedResponse> {
+  }: WebhookProcessParams): Promise<void> {
```

Comment:
Should we not be returning `AdapterResponse` here, as the outcome of `abstractConvertResponse`? Will this work on workers?

Reply:
🤦🏻‍♂️ Right, we have to return it for the Workers scenario